go/vt/vttablet/endtoend/misc_test.go

```diff
@@ -467,14 +467,41 @@ func TestQueryStats(t *testing.T) {
 		RowsReturned: 0,
 		ErrorCount:   1,
 	}
-	if stat != want {
-		t.Errorf("stat: %+v, want %+v", stat, want)
-	}
+	utils.MustMatch(t, want, stat)
 	vend := framework.DebugVars()
 	compareIntDiff(t, vend, "QueryCounts/vitess_a.Select", vstart, 2)
 	compareIntDiff(t, vend, "QueryRowCounts/vitess_a.Select", vstart, 0)
+	compareIntDiff(t, vend, "QueryRowsReturned/vitess_a.Select", vstart, 2)
 	compareIntDiff(t, vend, "QueryErrorCounts/vitess_a.Select", vstart, 1)
 
+	query = "update /* query_stats */ vitess_a set name = 'a'"
+	_, _ = client.Execute(query, bv)
+	defer func() {
+		// restore the table rows for other tests to use
+		query = "update /* query_stats */ vitess_a set name = 'abcd' where id = 1"
+		_, _ = client.Execute(query, bv)
+		query = "update /* query_stats */ vitess_a set name = 'bcde' where id = 2"
+		_, _ = client.Execute(query, bv)
+	}()
+	stat = framework.QueryStats()[query]
+	stat.Time = 0
+	stat.MysqlTime = 0
+	want = framework.QueryStat{
+		Query:        query,
+		Table:        "vitess_a",
+		Plan:         "UpdateLimit",
+		QueryCount:   1,
+		RowsAffected: 2,
+		RowsReturned: 0,
+		ErrorCount:   0,
+	}
+	utils.MustMatch(t, want, stat)
+	vend = framework.DebugVars()
+	compareIntDiff(t, vend, "QueryCounts/vitess_a.UpdateLimit", vstart, 1)
+	compareIntDiff(t, vend, "QueryRowCounts/vitess_a.UpdateLimit", vstart, 2)
+	compareIntDiff(t, vend, "QueryRowsAffected/vitess_a.UpdateLimit", vstart, 2)
+	compareIntDiff(t, vend, "QueryErrorCounts/vitess_a.UpdateLimit", vstart, 0)
```

Comment:
Similarly, we should assert here that the `QueryRowCounts` stat doesn't change.
Ideally we should also be testing `Delete` and `Insert` plans.

Reply:
I'll add tests for `Delete` and `Insert`. Its the same thing, `QueryRowCounts` will not be available for these types. I'll assert that it doesn't exist

Reply:
done